File: vdp-node/test/pymntacntattrinqry/generalattributesinquirytest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Payment Account Attributes General Inquiry Test', function() {
    var visaAPIClient = new VisaAPIClient();
    var generalAttributeInquiry = JSON.stringify({
          "primaryAccountNumber": "4465390000029077"
    });
    
    it('General Attributes Enquiry',function(done) {
        this.timeout(10000);
        var baseUri = 'paai/';
        var resourcePath = 'generalattinq/v1/cardattributes/generalinquiry';
        visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, generalAttributeInquiry, 'POST', {}, 
        function(err, responseCode) {
            if(!err) {
                assert.equal(responseCode, 200);
            } else {
                assert(false);
            }
            done();
        });    
    });
});

File: visa-dev-samplecode-master/vdp-ruby/test/pymntacntattrinqry/general_attributes_inquiry_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class GeneralAttributesInquiryTest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @generalAttributeInquiry ='''{
      "primaryAccountNumber": "4465390000029077"
    }'''
  end
  
  def test_fundsTransferInquiry
    base_uri = 'paai/'
    resource_path = 'generalattinq/v1/cardattributes/generalinquiry'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "General Attributes Inquiry test", "post", @generalAttributeInquiry)
    assert_equal("200", response_code, "General Attributes Inquiry test failed")
  end
end
